### PR TITLE
Support k8s list for init

### PR DESCRIPTION
**Pull request by Rob0h**
What I Did
------------
- Support k8s list for init

How I Did it
------------
- Split list yaml
- Store information on split into state.json
- Reassemble list yaml based on state.json

How to verify it
------------
- Integration
- run `ship init https://raw.githubusercontent.com/jaegertracing/jaeger-kubernetes/master/jaeger-production-template.yml`

Description for the Changelog
------------
Support k8s list for init

Picture of a Boat (not required but encouraged)
------------
🛶

### CODE REVIEW

File: pkg/specs/chart.go [hunk 1]
```diff
@@ -390,3 +376,91 @@ func (r *Resolver) maybeSplitMultidocYaml(ctx context.Context, localPath string)
...
+	if err != nil {
+		return errors.Wrapf(err, "read files in %s", path)
+	}
+
+	var lists []state.List
+	for _, file := range files {
+		filePath := filepath.Join(path, file.Name())
+
+		if file.IsDir() {
+			// TODO: handling nested list yamls
```
Comment (Rob0h) [inline pkg/specs/chart.go]: not sure of how important this is / how often it will come up

File: integration/init/jaeger/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+upstream: "https://github.com/jaegertracing/jaeger-kubernetes/blob/59e7afcae7b2f3bc109695cfda2a5c116cea4391/jaeger-production-template.yml"
+args: []
```
Comment (laverya) [inline integration/init/jaeger/metadata.yaml]:
```suggestion
args: ["--prefer-git"]
```
To help the GitHub API rate limits of people running the integration tests locally

File: pkg/specs/chart.go (hunk 1 above)
Comment (laverya) [reply pkg/specs/chart.go]: Perhaps call the function recursively and then `continue`? Either way, I know you need to add a `continue` here or things will blow up when it hits the ReadFile call below

File: pkg/specs/chart.go [hunk 3]
```diff
@@ -390,3 +376,91 @@ func (r *Resolver) maybeSplitMultidocYaml(ctx context.Context, localPath string)
...
 	return nil
 }
+
+type ListK8sYaml struct {
+	APIVersion string        `json:"apiVersion" yaml:"apiVersion"`
+	Kind       string        `json:"kind" yaml:"kind" hcl:"kind"`
+	Items      []interface{} `json:"items" yaml:"items"`
+}
+
+func (r *Resolver) maybeSplitListYaml(ctx context.Context, path string) error {
```
Comment (laverya) [inline pkg/specs/chart.go]: Could we get a test for this function? You should be able to largely copy https://github.com/replicatedhq/ship/blob/master/pkg/specs/chart_test.go#L125 for the test code

**Comment by laverya** [comment]
@Rob0h this look like a good idea for an integration test? https://github.com/replicatedhq/test-charts/pull/8

**Pull request merged**